Assets/Mirror/Runtime/InterestManagement/InterestManagement.cs

```diff
@@ -0,0 +1,133 @@
+                {
+                    // add it to observers
+                    identity.rebuild.Add(identity.connectionToClient);
+                }
+
+                // foreach rebuild
+                foreach (NetworkConnectionToClient conn in identity.rebuild)
+                {
+                    // was it not in old observers?
+                    if (!identity.observersx.Contains(conn))
+                    {
+                        // spawn identity for this connection
+                        //Debug.LogWarning($"Spawning {identity.name} for connectionId {conn.connectionId}");
+                        NetworkServer.ShowForConnection(identity, conn);
+
+                        // add it to observers
+                        identity.observersx.Add(conn);
+                    }
+                }
+            }
+        }
+
+        // rebuild all areas of interest for everyone once.
+        // the three rebuild steps are always the same. only RebuildObservers is
+        // different depending on the solution.
+        //
+        // note:
+        //   we DO NOT do any custom rebuilding after someone joined/spawned or
+        //   disconnected/unspawned.
+        //   this would require INSANE complexity.
+        //   for example, OnTransportDisconnect would have to:
+        //     1. remove the connection so the connectionId is invalid
+        //     2. then call BroadcastAfterUnspawn(oldEntity) which broadcasts
+        //        destroyed messages BEFORE rebuilding so we know the old
+        //        observers that need to get the destroyed message
+        //     3. RebuildAfterUnspawn to remove it
+        //     4. then remove the Entity from connection's owned objects, which
+        //        IS NOT POSSIBLE anymore because the connection was already
+        //        removed. which means that the next rebuild would still see it
+        //        etc.
+        //        (it's just insanity)
+        //   additionally, we would also need extra flags in Spawn to NOT
+        //   rebuild when spawning 10k scene objects in start, etc.s
+        //
+        // first principles:
+        //   it wouldn't even make sense to have special cases because players
+        //   might walk in and out of range from each other all the time anyway.
+        //   we already need to handle that case. (dis)connect is no different.
+        //
+        public void RebuildAll()
+        {
+            RebuildObservers();
+            RemoveOldObservers();
+            AddNewObservers();
+
+            // RebuildAll can be called from the outside, so we should set the
+            // lastUpdateTime in here instead of in Update.
+            // this way we avoid updating if someone from the outside recently
+            // did a rebuild anyway (e.g. in OnServerDisconnect).
+            lastUpdateTime = Time.time;
+        }
+
+        // update rebuilds every couple of seconds
+        void Update()
```

Comment:
This update should be optional, not all interest management need to be updated every x seconds. 

A Visibility script I use for one of my projects only rebuilds when an objects changes rooms.